### ISSUE TITLE
[Issue]: generating movie library thumbnails uses an crazy amount of memory

### ISSUE THREAD

**Issue by Thijzer** [jellyfin/jellyfin#6877]
### Please describe your bug

I noticed that when a movie library has to load in it's thumbnails by URL
<DOMAIN>/Items/<ID>/Images/Primary?fillHeight=342&fillWidth=228&quality=96&tag=0619c8b9bbbf8e088fb2f04be75864f0

- it needs to generate them on every new device, no cache pool?
- the generating of the thumbs reserves a lot of memory that doesn't get GB collected quickly

I'm willing to take a look into this, I just don't know why it has been setup this way, if this behavior is expected?

Can someone please give me some hints.

Thanks

### Jellyfin Version

10.7.7

### Environment

```markdown
- OS: 
- Virtualization: docker
- Clients: Brave Browser
- Browser:
- FFmpeg Version:
- Playback Method:
- Hardware Acceleration:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage: via docker volumes native ext4
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cvium** [comment]
Define a lot

**Comment by hannesbraun** [comment]
I've had a similar experience using Jellyfin 10.8.4 inside a Docker container running on my Raspberry Pi 2. When the thumbnails are requested and aren't cached yet, I get a huge spike in memory usage that usually ends in Jellyfin crashing because the system's out of memory. To put it in numbers, my memory usage is about 340 MB when Jellyfin is running idle. During those requests, I can see values of up to 800 MB. (I'd guess it's even higher, but the refresh rate of my resource monitor is probably too slow for that.) The thumbnail page contained about 60 images.

> it needs to generate them on every new device, no cache pool?

As far as I was able to understand the current implementation, there is a cache pool. But different devices may require different resolutions. Thus, the images need to be resized.

> the generating of the thumbs reserves a lot of memory that doesn't get GB collected quickly

In the rare cases where Jellyfin didn't crash, the memory got garbage collected rather quickly, however. I didn't notice any significant change in "memory idle usage" after about 2 seconds or something like that.

From what I tried, limiting the number of parallel image encoding tasks should help to avoid those "big" memory spikes.
My suggestion would be to introduce an environment variable or another kind of configuration parameter that lets you limit the number of parallel image execution tasks in case you're running short of memory. I think, although you're sacrificing speed, at least you can prevent the server from crashing. I do have a working implementation for this. When enabled, it limits the number of parallel encoding tasks to the number of the processors available in the system, as this may be a good tradeoff between speed and memory usage. For me, this limits the memory usage during image encoding to a maximum of 600 MB. If you consider this to be useful, I'm happy to submit my change as a pull request. What do you think?

**Comment by Bond-009** [comment]
Should be fixed by #8783
